### ISSUE TITLE
Training on Multiple GPUs

### ISSUE THREAD

**Issue by eric2579** [UKPLab/sentence-transformers#198]
Hi, I was able to train on GPU successfully. I was wondering if the package supports training on multiple GPUs? If not, is there a way for me to modify the source code to take advantage of training in parallel?

Thank you.

**Comment by nreimers** [comment]
Hi @eric2579 
So far it is not supported. I did some experiments, but training on multiple GPU sadly slowed down the training process. So it is a bit more complex than just using the DataParallel wrapper from pytorch.

If you find a way, please let me know.

**Comment by eric2579** [comment]
Hi @nreimers 
Thank you for your quick response. Will do.
Even though it yielded poor performance, do you mind sharing how you went about and altered the source code to use DataParallel? This may be a good starting point for me to investigate.

**Comment by nreimers** [comment]
Hi @eric2579 
I sadly don't have that code any more. But I think I wrapped the train object model (like SoftMaxLoss) in DataParallel. But I'm not sure anymore.

**Comment by dr-aheydari** [comment]
@nreimers @eric2579 Hello! I am also very curious about parallelizing on multiple GPUs (for training or inference); any updates on training on multiple GPUs? it would be greatly appreciated (even if it is in the prototype phase).

Thank you very much.

**Comment by CallMeSp** [comment]
> @nreimers @eric2579 Hello! I am also very curious about parallelizing on multiple GPUs (for training or inference); any updates on training on multiple GPUs? it would be greatly appreciated (even if it is in the prototype phase).
> 
> Thank you very much.
> 
> **Edit:**
> 
> I added the `nn.DataParallel` wrapper to the `Transformer.py ` for the Auto_Model call and I did the same for the `RoBERTa.py` to the `RobertaModel.from_pretrained` (my two use cases) and it seemed to do the trick. I was able to get much faster fine-tuning time (on 8 16GB GPUs vs before on 1 16GB GPU).
> 
> Did you happen to try something similar and it didn't work?

do you mind sharing how you went about and altered the source code to use DataParallel?
